refactor: standardize test seed usage and remove deprecated test infrastructure

yilmazbahadir opened the pull request:
## 📝 Summary

This PR standardizes test seed usage across the e2e test suite and removes deprecated test infrastructure. All tests now use a centralized `TEST_SEED` constant derived from a standard test mnemonic, eliminating redundant seed derivation and improving test consistency.

## 🔧 Context / Implementation

**Problem:**
- Tests were using different methods to derive seeds (some from mnemonics, some from wallet jobs)
- Redundant seed derivation code scattered across multiple test files
- Deprecated `wallet-jobs.test.ts` and related utilities were still present but unused
- Inconsistent test data made it harder to maintain and debug tests

**Solution:**
- Updated `TEST_MNEMONIC` to use the standard "test test test..." mnemonic (compatible with Foundry)
- Added `TEST_SEED` constant in `testConstants.ts` to centralize seed derivation
- Updated all e2e tests (BTC, BLS, Solana, determinism) to use `TEST_SEED` instead of deriving seeds locally
- Removed deprecated `wallet-jobs.test.ts` (1059 lines) and related `initializeClient.ts` utility
- Cleaned up unused imports and helper functions from test utilities
- Removed `e2e-wj` script from package.json

**Key Changes:**
- `testConstants.ts`: Added `TEST_SEED` export, updated `TEST_MNEMONIC` to standard test mnemonic
- `btc.test.ts`: Updated to use `TEST_SEED` instead of wallet job seed export
- `bls.test.ts`: Replaced mnemonic-to-seed derivation with `TEST_SEED`, removed wallet seed retrieval test
- `determinism.test.ts`: Simplified to use `TEST_SEED` directly
- `solana/*.test.ts`: Updated all Solana tests to use `TEST_SEED`
- `runners.ts`: Removed unused EVM test helpers and `initializeSeed` dependency
- Deleted: `wallet-jobs.test.ts`, `initializeClient.ts`

## 🧪 Test Plan

**Steps to verify the change:**

1. Run the updated e2e test suites to ensure they all pass with the new test seed:
   pnpm e2e-btc
   pnpm e2e-sign-bls
   pnpm e2e-sign-determinism
   pnpm e2e-sign-solana
   2. Verify that tests are deterministic and produce consistent results across multiple runs

3. Confirm that the removed `e2e-wj` script is no longer available:sh
   pnpm e2e-wj  # Should fail with "unknown command"
   **Expected result:**
- All e2e tests pass successfully
- Tests use the centralized `TEST_SEED` constant
- No references to deprecated wallet-jobs test infrastructure
- Test output is consistent and deterministic

## 🖼️ Screenshots (if applicable)

| Before | After |
| ------ | ----- |
| Tests derived seeds individually from mnemonics or wallet jobs | All tests use centralized `TEST_SEED` constant |
| 1633 lines of code (including unused test files) | 115 insertions, 1633 deletions (net -1518 lines) |
| Multiple seed derivation methods | Single source of truth for test seed |

<!--

## ✅ Checklist

- [x] Follows **conventional commit** style (`feat:`, `fix:`, `chore:`, etc.)
- [ ] Tests / CI checks pass
- [ ] Reviewed and approved by required **CODEOWNERS**
- [ ] Branch is up-to-date and ready to merge

-->

yilmazbahadir committed:
test: update determinism to use test seed

yilmazbahadir committed:
test: update solana e2e tests to use test mnemonic

yilmazbahadir committed:
test: update runGeneric function to use test seed

yilmazbahadir committed:
test: use test seed for e2e-btc test

yilmazbahadir committed:
refactor: remove unused wallet-jobs test and update tests to use test seed

Remove deprecated test infrastructure:
- Delete wallet-jobs.test.ts and related e2e-wj script
- Remove initializeClient.ts utility (no longer used)
- Clean up unused imports and functions from test utilities
- Simplify test runners by removing unused EVM test helpers

Update tests to use test seed:
- Update BLS test to use TEST_SEED constant instead of deriving from mnemonic
- Remove wallet seed retrieval test from BLS suite

This cleanup removes 1378 lines of unused test code and standardizes test seed usage.

yilmazbahadir committed:
fix: lint

yilmazbahadir committed:
refactor: replace FOUNDRY_TEST_SEED with TEST_SEED

yilmazbahadir commented on src/__test__/e2e/signing/determinism.test.ts:
```diff
@@ -45,88 +38,31 @@
         0,
       ] as WalletPath;
-      const addr0 = '0x17E43083812d45040E4826D2f214601bc730F60C';
+      const addr0 = deriveAddress(TEST_SEED, path0);
       const path1 = [
         BTC_PURPOSE_P2PKH,
         ETH_COIN,
```
I think it's expected that a Device(HW or Simulator) should set the exact test mnemonic before they run the tests. For the device it's possible to set the test mnemonic to a safe card. And for the simulator it is actually the default one and also it can be set via env variable. So I think we're safe.

yilmazbahadir committed:
fix: lint

netbonus reviewed (approved):
great job! love how many deletions we got out of this.

we could update docs/readme with a callout to make a safecard using the expected mnemonic to run the tests.

also, might be worth adding comments or throwing errors in some tests if the mnemonic is wrong. it's really easy to lose a lot of time with things like this.

yilmazbahadir committed:
docs: add test mnemonic setup instructions and validation

Add SafeCard setup documentation with required test mnemonic.
Enhance test error messages to indicate incorrect mnemonic usage.
